doc/api/fs.md

```diff
@@ -1381,6 +1381,15 @@ the specified file descriptor. This means that no `'open'` event will be
 emitted. Note that `fd` should be blocking; non-blocking `fd`s should be passed
 to [`net.Socket`][].
 
+Also note that the blocking fd, if pointing to a character device (such as
+keyboard or sound card) can potentially block the main thread on stream close.
+This is because these devices do not produce EOF character as part of their data
+flow cycle, and thereby epitomize endless streams. As a result, they do not
+respond to `stream.close()` call. A workaround is to close the stream first
+using `stream.close()` and then push a random character into the stream.
+This unblocks the reader thread, leads to the completion of the data flow cycle,
+and the actual closing of the stream.
```

Comment:
I'm not sure I follow this last sentence.

Reply:
the problem with character device when abstracted as streams in Node is that they operate on a blocking file descriptor, managed by pooled worker threads, with a passback async handle to the main thread. so once they engage in reading, there is no way to unblock the thread, other than passing a dummy character to 'unblock' the reader thread, close the libuv handles and disengage from further reading.

Reply:
Is that something you can do with the fs api here? An example might work to reduce confusion :)

Reply:
added an example